sys/xtimer/xtimer_core.c

```diff
@@ -176,9 +166,20 @@ int _xtimer_set_absolute(xtimer_t *timer, uint32_t target)
     DEBUG("timer_set_absolute(): now=%" PRIu32 " target=%" PRIu32 "\n", now, target);
 
     timer->next = NULL;
-    if ((target >= now) && ((target - XTIMER_BACKOFF) < now)) {
+    /* if target is smaller or equal now shoot */
+    if(target <= now){
+        _shoot(timer);
+        return 0;
+    }
+    else if ((target - XTIMER_BACKOFF) < now) {
         /* backoff */
-        xtimer_spin_until(target + XTIMER_BACKOFF);
+        /* target - now can only be positiv */
+        now = _xtimer_now();
```

Comment:
Nnot any more as i removed the other check, this was here to compensate for the dealy from the previous check. Although i think it would not matter as the thread is spinning.